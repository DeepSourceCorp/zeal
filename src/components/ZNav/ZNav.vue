<template>
  <div class="z-nav gap-3 overflow-auto flex flex-nowrap">
    <slot></slot>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'

export default Vue.extend({
  name: 'ZTabList'
})
</script>
