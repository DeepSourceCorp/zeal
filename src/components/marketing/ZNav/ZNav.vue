<template>
  <nav
    class="sticky z-1000 top-0 lg:backdrop-blur bg-ink-400 lg:bg-opacity-25 text-vanilla-100 h-16 flex items-center"
  >
    <div class="flex items-center w-screen mx-auto" :class="[`${CONTAINERS[container].classes}`]">
      <div v-if="$slots.brand" class="flex items-center">
        <slot name="brand"></slot>
      </div>
      <span class="lg:static hidden lg:flex items-center lg:w-full">
        <div
          v-if="$slots['desktop-menu']"
          class="lg:flex flex-1 items-center mx-4"
          :class="[`${MENU_ITEMS_ALIGNMENT[menuItemsAlign].classes}`]"
        >
          <slot name="desktop-menu"></slot>
        </div>
        <div v-if="$slots['desktop-cta']" class="lg:flex items-center">
          <slot name="desktop-cta"></slot>
        </div>
      </span>
      <span @click="toggleHamburgerMenu()" class="lg:hidden ml-auto cursor-pointer">
        <z-icon icon="menu"></z-icon>
      </span>
    </div>
    <aside
      class="lg:hidden bg-ink-400 transform top-0 right-0 w-full fixed h-full overflow-auto ease-in-out transition-all duration-300 z-30"
      :class="[`${isHamburgerMenuVisible ? 'translate-x-0' : 'translate-x-full'}`]"
    >
      <span class="flex items-center justify-end px-4 h-16 border-b">
        <span @click="toggleHamburgerMenu()" class="cursor-pointer">
          <z-icon icon="x"></z-icon>
        </span>
      </span>
      <div
        v-if="$slots['mobile-menu']"
        class="lg:flex flex-1 items-center mx-4"
        :class="[`${MENU_ITEMS_ALIGNMENT[menuItemsAlign].classes}`]"
      >
        <slot name="mobile-menu"></slot>
      </div>
      <div v-if="$slots['mobile-cta']" class="lg:flex items-center">
        <slot name="mobile-cta"></slot>
      </div>
    </aside>
  </nav>
</template>

<script lang="ts">
import ZIcon from '@/components/ZIcon/ZIcon.vue'
import Vue from 'vue'

const CONTAINERS = {
  sm: { text: 'sm', classes: 'px-4 lg:px-0 lg:max-w-5xl' },
  md: { text: 'md', classes: 'px-4 lg:px-0 lg:max-w-6xl' },
  lg: { text: 'lg', classes: 'px-4 lg:px-0 lg:max-w-7xl' }
}

const MENU_ITEMS_ALIGNMENT = {
  left: { text: 'left', classes: 'justify-start' },
  center: { text: 'center', classes: 'justify-center' },
  right: { text: 'right', classes: 'justify-end' }
}

export default Vue.extend({
  components: { ZIcon },
  name: 'ZNav',
  props: {
    container: {
      type: String,
      default: CONTAINERS.md.text,
      validator: (container) => Object.keys(CONTAINERS).includes(container)
    },
    menuItemsAlign: {
      type: String,
      default: MENU_ITEMS_ALIGNMENT.center.text,
      validator: (align) => Object.keys(MENU_ITEMS_ALIGNMENT).includes(align)
    }
  },
  data() {
    return {
      isHamburgerMenuVisible: false,
      CONTAINERS,
      MENU_ITEMS_ALIGNMENT
    }
  },
  methods: {
    toggleHamburgerMenu() {
      this.isHamburgerMenuVisible = !this.isHamburgerMenuVisible
    }
  }
})
</script>
