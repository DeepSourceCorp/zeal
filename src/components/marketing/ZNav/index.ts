import ZNav from './ZNav.vue'

export default ZNav
