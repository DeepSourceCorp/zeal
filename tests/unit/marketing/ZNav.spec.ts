import Vue from 'vue'
import { createLocalVue, mount, Wrapper } from '@vue/test-utils'

import ZNav from '../../../src/components/marketing/ZNav'

describe('ZNavList component', () => {
  let mountFunction: (options?: Record<string, unknown>) => Wrapper<Vue>
  let localVue: typeof Vue
  beforeEach(() => {
    localVue = createLocalVue()
    mountFunction = (options = {}) => {
      return mount(ZNav, {
        localVue,
        ...options
      })
    }
  })
  it('renders a ZNav when the component is mounted', () => {
    expect(mountFunction().html()).toMatchSnapshot()
  })
  it('renders a ZNav with proper container size', () => {
    let mountedInstance = mountFunction({
      propsData: {
        container: 'md'
      }
    })
    expect(mountedInstance.html()).toMatchSnapshot()
    expect(mountedInstance.html()).toContain('lg:max-w-6xl')
  })
  it('renders a ZNav with proper menu alignment', () => {
    let mountedInstance = mountFunction({
      propsData: {
        menuAlign: 'right'
      }
    })
    expect(mountedInstance.html()).toMatchSnapshot()
    expect(mountedInstance.html()).toContain('justify-end')
  })
  it('renders a ZNav with drawer to toggle', async () => {
    let mountedInstance = mountFunction({
      propsData: {
        menuAlign: 'right'
      }
    })
    expect(mountedInstance.html()).toMatchSnapshot()
    await mountedInstance.find('.lg\\:hidden.cursor-pointer').trigger('click')
    expect(mountedInstance.html()).toContain('translate-x-0')
  })
})
