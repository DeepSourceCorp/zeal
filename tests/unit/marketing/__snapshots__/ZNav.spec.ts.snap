// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ZNavList component renders a ZNav when the component is mounted 1`] = `
<nav class="z-1000 sticky flex items-center bg-ink-400 text-vanilla-100 top-0 border-b border-ink-200 h-16 lg:backdrop-blur lg:bg-opacity-25 lg:border-0">
  <div class="flex items-center w-screen lg:mx-auto px-4 lg:max-w-6xl">
    <!----> <span class="hidden lg:flex items-center w-full"><!----> <!----></span> <span class="lg:hidden ml-auto cursor-pointer"><svg viewBox="0 0 24 24" stroke-linecap="round" stroke-linejoin="round" fill="none" class="z-icon stroke-1.5 w-6 h-6 z-icon--menu text-vanilla-400"><line x1="3" y1="12" x2="21" y2="12"></line><line x1="3" y1="6" x2="21" y2="6"></line><line x1="3" y1="18" x2="21" y2="18"></line></svg></span>
  </div>
  <aside class="fixed lg:hidden bg-ink-300 transform top-0 right-0 w-full h-full overflow-auto ease-in-out transition-all duration-300 z-30 translate-x-full"><span class="flex items-center justify-end border-b border-ink-200 h-16 px-4"><span class="cursor-pointer"><svg viewBox="0 0 24 24" stroke-linecap="round" stroke-linejoin="round" fill="none" class="z-icon stroke-1.5 w-6 h-6 z-icon--x text-vanilla-400"><line x1="18" y1="6" x2="6" y2="18"></line><line x1="6" y1="6" x2="18" y2="18"></line></svg></span></span>
    <!---->
    <!---->
  </aside>
</nav>
`;

exports[`ZNavList component renders a ZNav with drawer to toggle 1`] = `
<nav class="z-1000 sticky flex items-center bg-ink-400 text-vanilla-100 top-0 border-b border-ink-200 h-16 lg:backdrop-blur lg:bg-opacity-25 lg:border-0">
  <div class="flex items-center w-screen lg:mx-auto px-4 lg:max-w-6xl">
    <!----> <span class="hidden lg:flex items-center w-full"><!----> <!----></span> <span class="lg:hidden ml-auto cursor-pointer"><svg viewBox="0 0 24 24" stroke-linecap="round" stroke-linejoin="round" fill="none" class="z-icon stroke-1.5 w-6 h-6 z-icon--menu text-vanilla-400"><line x1="3" y1="12" x2="21" y2="12"></line><line x1="3" y1="6" x2="21" y2="6"></line><line x1="3" y1="18" x2="21" y2="18"></line></svg></span>
  </div>
  <aside class="fixed lg:hidden bg-ink-300 transform top-0 right-0 w-full h-full overflow-auto ease-in-out transition-all duration-300 z-30 translate-x-full"><span class="flex items-center justify-end border-b border-ink-200 h-16 px-4"><span class="cursor-pointer"><svg viewBox="0 0 24 24" stroke-linecap="round" stroke-linejoin="round" fill="none" class="z-icon stroke-1.5 w-6 h-6 z-icon--x text-vanilla-400"><line x1="18" y1="6" x2="6" y2="18"></line><line x1="6" y1="6" x2="18" y2="18"></line></svg></span></span>
    <!---->
    <!---->
  </aside>
</nav>
`;

exports[`ZNavList component renders a ZNav with proper container size 1`] = `
<nav class="z-1000 sticky flex items-center bg-ink-400 text-vanilla-100 top-0 border-b border-ink-200 h-16 lg:backdrop-blur lg:bg-opacity-25 lg:border-0">
  <div class="flex items-center w-screen lg:mx-auto px-4 lg:max-w-6xl">
    <!----> <span class="hidden lg:flex items-center w-full"><!----> <!----></span> <span class="lg:hidden ml-auto cursor-pointer"><svg viewBox="0 0 24 24" stroke-linecap="round" stroke-linejoin="round" fill="none" class="z-icon stroke-1.5 w-6 h-6 z-icon--menu text-vanilla-400"><line x1="3" y1="12" x2="21" y2="12"></line><line x1="3" y1="6" x2="21" y2="6"></line><line x1="3" y1="18" x2="21" y2="18"></line></svg></span>
  </div>
  <aside class="fixed lg:hidden bg-ink-300 transform top-0 right-0 w-full h-full overflow-auto ease-in-out transition-all duration-300 z-30 translate-x-full"><span class="flex items-center justify-end border-b border-ink-200 h-16 px-4"><span class="cursor-pointer"><svg viewBox="0 0 24 24" stroke-linecap="round" stroke-linejoin="round" fill="none" class="z-icon stroke-1.5 w-6 h-6 z-icon--x text-vanilla-400"><line x1="18" y1="6" x2="6" y2="18"></line><line x1="6" y1="6" x2="18" y2="18"></line></svg></span></span>
    <!---->
    <!---->
  </aside>
</nav>
`;

exports[`ZNavList component renders a ZNav with proper menu alignment 1`] = `
<nav class="z-1000 sticky flex items-center bg-ink-400 text-vanilla-100 top-0 border-b border-ink-200 h-16 lg:backdrop-blur lg:bg-opacity-25 lg:border-0">
  <div class="flex items-center w-screen lg:mx-auto px-4 lg:max-w-6xl">
    <!----> <span class="hidden lg:flex items-center w-full"><!----> <!----></span> <span class="lg:hidden ml-auto cursor-pointer"><svg viewBox="0 0 24 24" stroke-linecap="round" stroke-linejoin="round" fill="none" class="z-icon stroke-1.5 w-6 h-6 z-icon--menu text-vanilla-400"><line x1="3" y1="12" x2="21" y2="12"></line><line x1="3" y1="6" x2="21" y2="6"></line><line x1="3" y1="18" x2="21" y2="18"></line></svg></span>
  </div>
  <aside class="fixed lg:hidden bg-ink-300 transform top-0 right-0 w-full h-full overflow-auto ease-in-out transition-all duration-300 z-30 translate-x-full"><span class="flex items-center justify-end border-b border-ink-200 h-16 px-4"><span class="cursor-pointer"><svg viewBox="0 0 24 24" stroke-linecap="round" stroke-linejoin="round" fill="none" class="z-icon stroke-1.5 w-6 h-6 z-icon--x text-vanilla-400"><line x1="18" y1="6" x2="6" y2="18"></line><line x1="6" y1="6" x2="18" y2="18"></line></svg></span></span>
    <!---->
    <!---->
  </aside>
</nav>
`;
